pkg/fanal/artifact/local/fs.go

```diff
@@ -58,17 +63,85 @@ func NewArtifact(rootPath string, c cache.ArtifactCache, w Walker, opt artifact.
 		return nil, xerrors.Errorf("analyzer group error: %w", err)
 	}
 
-	return Artifact{
+	art := Artifact{
 		rootPath:       filepath.ToSlash(filepath.Clean(rootPath)),
+		logger:         log.WithPrefix("fs"),
 		cache:          c,
 		walker:         w,
 		analyzer:       a,
 		handlerManager: handlerManager,
 		artifactOption: opt,
-	}, nil
+	}
+
+	// Check if the directory is a git repository and clean
+	if hash, err := gitCommitHash(art.rootPath); err == nil {
+		art.logger.Debug("Using the latest commit hash for calculating cache key", log.String("commit_hash", hash))
+		art.commitHash = hash
+	} else {
+		art.logger.Debug("Random cache key will be used", log.Err(err))
```

Comment:
Looks like we don't need to show this log unless it's a git repository directory.
Because we will show the following log for each `fs` scan
```
[fs] Random cache key will be used      err="failed to open git repository: repository does not exist"
```

Reply:
also, it might make sense to add `path` here.

Reply:
Updated https://github.com/aquasecurity/trivy/pull/8278/commits/38021244dd7da23879b1ece5e4bdf54665525d13